src/OcisResource.php

```diff
@@ -402,4 +388,20 @@ public function getCheckSums(): array
         }
         return [];
     }
+
+    /**
+     * Returns the private link to the resource.
+     * This link can be used by any user with the correct permissions to navigate to the resource in the web UI.
+     * The link is urldecoded.
+     * @return string
+     * @throws InvalidResponseException
+     */
+    public function getPrivatelink(): string
+    {
+        $privateLink = $this->getMetadata(ResourceMetadata::PRIVATELINK);
+        if (is_array($privateLink)) {
```

Comment:
How does this happen - I guess that if the resource does not have any private link, then there is an empty array at this point? Maybe add a comment?

Reply:
this really should not happen, but the analyzer don't know, so I've made it more explicit by throwing an exception